### PR TITLE
Web: Remove Django webinterface

### PR DESCRIPTION
**Pull request by SchoolGuy**
We decided in <https://github.com/cobbler/cobbler/issues/2342> that we deprecate the Webinterface. Currently the webinterface is preventing me from introducing changes to Cobbler which are addressing tech-debt. Thus I would like to remove the webinterface and add a new repository as soon as the new API is ready. This means that we will have for future versions of Cobbler no Webinterface. I will try to keep the timeframe as short as possible but I can't say how long this will take exactly.

Upgrading to a non EOL Django version does not make a difference and is thus not something I will put any minute into.

This PR will not address any tech-debt which is blocking further refactoring and moderinzation of Cobbler.

### CODE REVIEW

**Comment by brejoc** [comment]
@SchoolGuy Looks good so far. Is there still something missing

**Comment by SchoolGuy** [comment]
@brejoc Not really. But I want to wait with merging until we are sure that 3.2.0 is mostly okay so we don't need to let the two branches diverge. Backporting fixes will be easier also with this not merged. And it mostly doesn't hurt for now. I will merge it when I got the approvals and I am getting started on working to redesign the internals.

**Comment by SchoolGuy** [comment]
So the remaining this to investigate is why RH based distros do fail. Tests on SUSEish distros seem to pass.

**Review by brejoc — CHANGES_REQUESTED** [review]
I think we can also remove some of those! Or am I mistaken?

```
$ grep -rin "django" *
docker/Fedora33.dockerfile:36:    python3-django          \
docker/CentOS8.dockerfile:49:    python3-django            \
docker/openSUSE_Leap.dockerfile:29:RUN ["pip3", "install", "pytest-django"]
docker/openSUSE_Leap.dockerfile:44:# set SECRET_KEY for django tests
docker/openSUSE_TW.dockerfile:28:RUN ["pip3", "install", "pytest-django"]
docker/openSUSE_TW.dockerfile:43:# set SECRET_KEY for django tests
docker/Debian10.dockerfile:25:    python-django-common \
docker/Debian10.dockerfile:31:    python3-django \
docs/requirements.rtd.txt:3:django
docs/installation-guide.rst:51:- Django / python-django
docs/installation-guide.rst:138:    $ apt-get -y install python3-django
SECURITY.md:39:Cobbler uses currently a very old Django version (for the web frontend) and the code of Cobbler itself is currently in
tests/build-and-install-rpms.sh:51:    docker exec -it cobbler bash -c 'pip3 install pyyaml simplejson netaddr Cheetah3 Django pymongo distro ldap3 librepo'
tests/build-and-install-debs.sh:50:    docker exec -it cobbler bash -c 'pip3 install pyyaml simplejson netaddr Cheetah3 Django pymongo distro ldap3 librepo'
tests/install-debs.sh:29:    docker exec -it cobbler bash -c 'pip3 install pyyaml simplejson netaddr Cheetah3 Django pymongo distro ldap3 librepo'
tests/install-rpms.sh:29:    docker exec -it cobbler bash -c 'pip3 install pyyaml simplejson netaddr Cheetah3 Django pymongo distro ldap3'
```

**Comment by SchoolGuy** [comment]
@brejoc Re-Review please. :)

File: setup.py
Comment (nodeg) [inline setup.py:557]: What about those two? Then line 351 has to be adjusted, too.

**Review by nodeg — APPROVED** [review]
I found some `webui` related content. Some of the content is a false positive but I think some of them could be removed.

```
$ grep -rin "webui" .*
.gitignore:29:cobbler/webui/master.py

$ grep -rin "webui" * 
cobbler/remote.py:308:            # NOTE: WebUI passes in repos here, CLI passes only:
cobbler/remote.py:2971:        Wrapper for ``utils.get_random_mac()``. Used in the webui.
Binary file cobbler/__pycache__/settings.cpython-39.pyc matches
Binary file cobbler/__pycache__/remote.cpython-39.pyc matches
cobbler/settings.py:145:                         "repo_mirror", "repo_profile", "repo_system", "svc", "web", "webui"],
cobbler/modules/authentication/denyall.py:3:Used to disable the WebUI by default.
cobbler.spec:453:%exclude %{_sharedstatedir}/cobbler/webui_sessions
config/cobbler/settings.yaml:161:# LDAP for WebUI/XMLRPC authentication.
config/cobbler/settings.yaml:377:  - webui
config/cobbler/users.conf:1:# Cobbler WebUI / Web Services authorization config file
config/cobbler/modules.conf:5:# Decides what users can log into the WebUI and call Read-Write XMLRPC methods.
config/cobbler/modules.conf:30:# once a user has been cleared by the WebUI/XMLRPC, what can they do?
docs/index.rst:13:mirrors with automated installation files. Cobbler has a command line interface, WebUI, and extensive Python and
docs/cobblerd.rst:38:Cobbler WebUI which is usable for day-to-day operation of Cobbler once installed/configured. Docs on the API and XML-RPC
docs/cobbler.rst:170:|                 | Cobbler users can edit particular objects.This only applies to the Cobbler WebUI and XML-RPC        |
docs/cobbler.rst:257:|                     | Cobbler users can edit particular objects.This only applies to the Cobbler WebUI and XML-RPC   |
docs/cobbler.rst:596:|                     | Cobbler users can edit particular objects.This only applies to the Cobbler WebUI and XML-RPC   |
docs/cobbler.rst:746:|                  | Cobbler users can edit particular objects.This only applies to the Cobbler WebUI and XML-RPC      |
docs/cobbler-conf.rst:416:using LDAP for WebUI/XML-RPC authentication.
docs/cobbler-conf.rst:849:      - webui
docs/cobbler-conf.rst:906:What users can log into the WebUI and Read-Write XML-RPC?
docs/cobbler-conf.rst:933:Once a user has been cleared by the WebUI/XML-RPC, what can they do?
scripts/settings-migration-v1-to-v2.sh:15:# what users can log into the WebUI and Read-Write XMLRPC?
scripts/settings-migration-v1-to-v2.sh:36:# once a user has been cleared by the WebUI/XMLRPC, what can they do?
setup.py:349:        path = os.path.join(self.root + libpath, 'webui_sessions')
setup.py:557:            ("%s/webui_sessions" % libpath, []),
tests/test_data/settings_old:176:# LDAP for WebUI/XMLRPC authentication.
tests/test_data/settings_old:395:  - webui
```

Furthermore some more django related content in `.bandit`.
```shell
$ grep -rin "django" .*
.bandit:76:# B610 : django_extra_used
.bandit:77:# B611 : django_rawsql_used
.bandit:80:# B703 : django_mark_safe
```

Otherwise LGTM.

File: setup.py
Comment (nodeg) [inline setup.py:557]:
```suggestion
            ("share/cobbler/web/templates", glob("web/templates/*")),
            ("%s/webui_sessions" % libpath, []),
```
What about those two? Then line 351 has to be adjusted, too.

**Review by brejoc — APPROVED** [review]
Looks good! :+1:

File: setup.py
Comment (SchoolGuy) [reply setup.py:557]: I am not sure. Let's leave it like this and see after some refactorings if that is needed or not.

**Pull request merged**
